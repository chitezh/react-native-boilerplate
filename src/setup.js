import React, { Component } from 'react';
import { Provider } from 'react-redux';

import App from './App';
import Splash from './components/scenes/splash/Splash';
import configureStore from './store';

export default () => {
  class Root extends Component {
    constructor() {
      super();
      this.state = {
        isLoading: true,
        store: configureStore(() => this.setState({ isLoading: false })),
      };
    }

    render() {
      if (this.state.isLoading) {
        return <Splash />;
      }

      return (
        <Provider store={this.state.store}>
          <App />
        </Provider>
      );
    }
    }
  return Root;
};
