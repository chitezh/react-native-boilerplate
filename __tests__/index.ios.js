import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import setup from '../src/setup';

const Index = setup();

it('renders correctly', () => {
  const tree = renderer.create(<Index />);
  expect(tree).toMatchSnapshot();
});
