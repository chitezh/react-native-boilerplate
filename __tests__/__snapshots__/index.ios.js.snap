// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-around",
      "padding": 50,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  />
  <ActivityIndicator
    animating={true}
    color="#2dad9c"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
